src/raftstore/store/peer_storage.rs

```diff
@@ -994,54 +1055,79 @@ impl PeerStorage {
 }
 
 /// Delete all meta belong to the region. Results are stored in `wb`.
-pub fn clear_meta(engine: &DB,
-                  wb: &WriteBatch,
+pub fn clear_meta(wb: &WriteBatch,
+                  raft_wb: &WriteBatch,
+                  raft_engine: &DB,
                   region_id: u64,
-                  raft_state: &RaftLocalState)
+                  raft_state: &RaftLocalState,
+                  is_stale_meta: bool)
                   -> Result<()> {
     let t = Instant::now();
     try!(wb.delete(&keys::region_state_key(region_id)));
+    try!(wb.delete(&keys::apply_state_key(region_id)));
+
+    let raft_count = if is_stale_meta {
```

Comment:
any difference between clearing all raft logs clearing from first to last raft logs?